verilog/tools/kythe/scope_resolver.h

```diff
@@ -0,0 +1,199 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_KYTHE_SCOPE_RESOLVER_H_
+#define VERIBLE_VERILOG_TOOLS_KYTHE_SCOPE_RESOLVER_H_
+
+#include <map>
+#include <vector>
+
+#include "absl/strings/string_view.h"
+#include "common/util/auto_pop_stack.h"
+#include "common/util/iterator_range.h"
+#include "verilog/tools/kythe/kythe_facts.h"
+
+namespace verilog {
+namespace kythe {
+
+// Used to wrap whatever needs to be recorded in a scope item.
+struct ScopeMemberItem {
+  ScopeMemberItem(const VName& vname) : vname(vname) {}
+
+  // VName of this member.
+  VName vname;
+};
+
+// Represents some scope in SystemVerilog code like class or module.
+class Scope {
+ public:
+  Scope() {}
```

Comment:
Prefer `Scope() = default;` for default constructors.

If it doesn't make sense to have a default contructor, consider `= delete;` to require one of the other constructors to guarantee proper initialization.  Ask: Should one be allowed to create a Scope without a Signature?

Reply:
This one is used for the Map